### PR TITLE
Feature 71: Add Source Download Button to page

### PR DESCRIPTION
**Pull request by PhotoNomad0**
Issue: https://github.com/unfoldingWord-dev/door43.org/issues/71

Changes
- Added Download button to bible and obs templates.
- Added Javascript to download button to open preconvert zip file
- modified project-page.js to save the source zip path if in build_log.json.  Otherwise it is generated from commit hash in URL.
- Added jasmine test framework for JS testing.

### CODE REVIEW

**Comment by PhotoNomad0** [comment]
@phillip-hopper - OK pulled out jasmine folder.  Got stuck in weeds trying to get karma set up, so will implement that later.

**Comment by phillip-hopper** [comment]
@PhotoNomad0 Github says the jasmine files are still in the PR.

**Comment by PhotoNomad0** [comment]
This is still being worked on to update buttons.

**Comment by PhotoNomad0** [comment]
@richmahn - thanks for the example.  How does this look to you?

File: test/spec/SpecBadges.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
+describe('Test Badges', function () {
```
Comment (richmahn) [inline test/spec/SpecBadges.js]: @PhotoNomad0 Why were these new files added?

**Comment by PhotoNomad0** [comment]
@phillip-hopper - I want to talk about this at standup.  It seems like Rich doesn't even think they should be there.  Also I got stuck in the weeds trying to convert SpecRunner.html to run on Karma with jQuery.  May need help with that.

**Comment by PhotoNomad0** [comment]
@phillip-hopper @richmahn - OK I have run the gauntlet and got the unit tests to run on Travis.  Anything else missing here?

**Comment by richmahn** [comment]
@phillip-hopper : Will let you verify

**Comment by PhotoNomad0** [comment]
Looks like the final stage is to get coveralls support.

**Pull request merged**